src/Hamiltonians/FroelichPolaron_nD.jl

```diff
@@ -0,0 +1,205 @@
+using SpecialFunctions
+
+#structure 
+struct FroehlichPolaron_nD{
+    T,
+    M,
+    D,
+    A<:OccupationNumberFS{M}, 
+    MC,
+    G<:CubicGrid
+    } <: AbstractHamiltonian{T}
+    address::A
+    d:: Int64
+    geometry::G
+    alpha::T
+    mass::T
+    omega::T
+    l::SVector{D, Float64}
+    p::SVector{D, Float64}
+    ks::SVector{M, SVector{D, T}}
+    momentum_cutoff::Union{MC, Nothing}
+    mode_cutoff::Union{Int, Nothing}
+end
+
+#function for nD Froehlich Polaron
+function FroehlichPolaron_nD(
+    address::OccupationNumberFS{M,AT};
+    D::Int = 2,
+    geometry::CubicGrid = PeriodicBoundaries(ntuple(_ -> Int(round(num_modes(address)^(1/D))), D)),
+    alpha = 1,
+    mass = 1,
+    omega = 1,
+    l = ones(D),
+    p = zeros(D),
+    momentum_cutoff::Union{Nothing,Real} = nothing,
+    mode_cutoff::Union{Nothing,Int} = nothing,
+) where {M, AT}
+
+    # Check positive l
+    if length(l) != D || any(x -> x <= 0, l)
+        throw(ArgumentError("`l` must be a positive vector of length $D"))
+    end
+
+    #check m_lin is possible
+    if abs(M^(1/D) - round(M^(1/D))) >0.01  
+        throw(ArgumentError("M = $M is not a perfect $D th power"))
+    end
+
+    alpha, mass, omega = promote(float(alpha), float(mass), float(omega))
+    l = SVector{D,Float64}(float.(l))
+    p = SVector{D,Float64}(float.(p))
+
+    ks_tmp = Vector{SVector{D,Float64}}(undef, M)
+
+    #build k grid
+    for idx_mode in 1:M
+    idx = Tuple(geometry[idx_mode])
+    kv = ntuple(d -> begin
+        m = idx[d] - 1 - div(round(M^(1/D)),2)   
+        (2π / l[d]) * m
+    end, D)
+    ks_tmp[idx_mode] = SVector{D,Float64}(kv)
+    end
```

Comment:
```suggestion
    for idx_mode in 1:M
        idx = Tuple(geometry[idx_mode])
        ntuple(D) do d
            m = idx[d] - 1 - div(round(M^(1/D)), 2)
            (2π / l[d]) * m
        end
        ks_tmp[idx_mode] = SVector{D,Float64}(kv)
    end
```
Please make sure the code is indented properly! The do notation is also helpful here.